test/extended/openstack/utils.go

```diff
@@ -122,3 +133,30 @@ func difference(a []string, b []string) []string {
 	}
 	return diff
 }
+
+func GetMachinesetRetry(client runtimeclient.Client, ms *machinev1.MachineSet, shouldExist bool) error {
+	var err error
+	const maxRetries = 5
+	const delay = 10
+	retries := 1
+	for retries < maxRetries {
+		_, err = framework.GetMachineSet(client, ms.Name)
+
+		if shouldExist {
+			if err != nil {
+				retries += 1
+				time.Sleep(time.Second * delay)
+			} else {
+				break
+			}
+		} else {
+			if err == nil {
+				retries += 1
+				time.Sleep(time.Second * delay)
+			} else {
+				break
+			}
+		}
```

Comment:
Whoa, what a code duplication,. why? How about:

	if err != nil == shouldExist {
		retries += 1
		time.Sleep(time.Second * delay)
	} else {
		break
	}

Please double-check me if the logic is identical.

You should also check the error type here if it's NotFound.

Reply:
agree! And logic is ok

Reply:
About the Notfound - I'm checking it in the test itself, do you think I should add the logic in the function?

Reply:
Nah, okay, I see NotFound check now.